frame/staking/src/lib.rs

```diff
@@ -1485,118 +2060,204 @@ impl<T: Trait> Module<T> {
 		})
 	}
 
-	/// Select a new validator set from the assembled stakers and their role preferences.
+	/// Select the new validator set at the end of the era.
 	///
-	/// Returns the new `SlotStake` value and a set of newly selected _stash_ IDs.
+	/// Runs [`try_do_phragmen`] and updates the following storage items:
+	/// - [`EraElectionStatus`]: with `None`.
+	/// - [`Stakers`]: with the new staker set.
+	/// - [`SlotStake`]: with the new slot stake.
+	/// - [`CurrentElected`]: with the new elected set.
+	/// - [`SnapshotValidators`] and [`SnapshotNominators`] are both removed.
 	///
-	/// Assumes storage is coherent with the declaration.
-	fn select_validators() -> (BalanceOf<T>, Option<Vec<T::AccountId>>) {
-		let mut all_nominators: Vec<(T::AccountId, Vec<T::AccountId>)> = Vec::new();
-		let all_validator_candidates_iter = <Validators<T>>::enumerate();
-		let all_validators = all_validator_candidates_iter.map(|(who, _pref)| {
-			let self_vote = (who.clone(), vec![who.clone()]);
-			all_nominators.push(self_vote);
-			who
-		}).collect::<Vec<T::AccountId>>();
+	/// Internally, [`QueuedElected`], snapshots and [`QueuedScore`] are also consumed.
+	///
+	/// If the election has been successful, It passes the new set upwards.
+	///
+	/// This should only be called at the end of an era.
+	fn select_and_update_validators() -> Option<Vec<T::AccountId>> {
+		if let Some(ElectionResult::<T::AccountId, BalanceOf<T>> {
+			elected_stashes,
+			exposures,
+			compute,
+		}) = Self::try_do_phragmen() {
```

Comment:
so if a solution has been submitted it is necessarily a submitted solution that will be choose

what about this situation:
* I'm validator with a good amount of nominators
* I submit my solution which is good, and then I just go chill,
* no following solution can use my nominators vote to me, and maybe there is no better solution without me.
* thus my proposition is very likely to be accepted.

Is this an issue ? I mean the submitted solution could be specifically crafted to avoid some specific people or unfair in some way I'm not aware of ?

There is probably nothing wrong there, but I kind of feel that being able to chill once I'm inside a solution can have some potential flaws maybe ?

Reply:
nope as you said, you can't chill once its open